src/OVHSwiftAdapter.php

```diff
@@ -83,8 +84,14 @@ public function getTemporaryUrl($path, $expiration, $options = []): string
     {
         $this->checkParams();
 
+        $path = $this->applyPathPrefix($path);
+
         // expiry is relative to current time
-        $expiresAt = $expiration instanceof Carbon ? $expiration->timestamp : (int) (time() + 60 * 60);
+        $expiresAt = $expiration;
```

Comment:
You’re breaking the timestamp implementation. Make sure to properly apply time arithmetics, as they’re needed for OVH’s timestamp validation.